### PR TITLE
Implement lcom4 cohesion metrics

### PR DESCRIPTION
**Pull request by fummicc1**
# Pull Request

## Overview

Implement LCOM4 cohesion metrics calculator. (linux is not supported)

## Type of Change

Please check all that apply:

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes)
- [ ] 🧪 Test improvements
- [ ] ⚡ Performance improvements
- [ ] 🔨 Build/CI improvements
- [ ] 🌍 Cross-platform compatibility changes

## Related Issues

Closes #(issue number)
Related to #(issue number)

### Core Changes

-
-

### Tests

-
-

### Documentation

-
-

### Test Plan

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Complexity analysis on self passes

### Test Commands

```bash
# Commands used to test this change
swift test
swift run swift-complexity Sources --threshold 15
```

### Self-Analysis Required

- [ ] Complexity analysis run on changed files
- [ ] No functions exceed threshold (15)
- [ ] New complex code is justified and documented

Run swift-complexity on your changes:

```bash
# Analyze changed files for complexity
swift run swift-complexity [changed-files] --threshold 15 --format text

# For comprehensive analysis
swift run swift-complexity Sources --threshold 15 --recursive --format json
```

**Results**: (paste results here or indicate no high-complexity functions added)

**Justification for complex functions**: (if any functions exceed threshold)

### Platform Testing

- [ ] Changes tested on macOS (Apple Silicon)
- [ ] Changes tested on macOS (Intel)
- [ ] Changes tested on Linux (Ubuntu 22.04+)
- [ ] Platform-specific code properly conditionally compiled

### Platform Considerations

- [ ] No platform-specific dependencies added
- [ ] Linux compatibility maintained
- [ ] macOS-specific features properly isolated (e.g., swift-format)

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance regression (with justification)

**Details**: (if applicable)

## Breaking Changes

- [ ] No breaking changes
- [ ] Breaking changes (describe below)

**Breaking Changes Description**: (if applicable)

## Screenshots/Output

(If applicable, add screenshots or example output)

#### Input

```
❯ .build/debug/SwiftComplexityCLI Sources/SwiftComplexityCore --lcom4 --project-root . --recursive 2>&1 | head -50
``` 

#### Output

<details><summary>Output</summary>
<p>

```
File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/CognitiveComplexityCalculator.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| CognitiveComplexityCalcul | class  | 2     | 12      | 4          | moderate   |
+---------------------------+--------+-------+---------+------------+------------+
Total: 1 classes, Average LCOM4: 2.00, Low cohesion: 0

File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/ComplexityAnalyzer.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| ComplexityAnalyzer        | actor  | 1     | 3       | 6          | high       |
+---------------------------+--------+-------+---------+------------+------------+
Total: 1 classes, Average LCOM4: 1.00, Low cohesion: 0

File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/CyclomaticComplexityCalculator.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| CyclomaticComplexityCalcu | class  | 1     | 11      | 1          | high       |
+---------------------------+--------+-------+---------+------------+------------+
Total: 1 classes, Average LCOM4: 1.00, Low cohesion: 0

File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/FunctionDetector.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| DetectedFunction          | struct | 1     | 1       | 4          | high       |
| FunctionDetector          | class  | 1     | 10      | 2          | high       |
+---------------------------+--------+-------+---------+------------+------------+
Total: 2 classes, Average LCOM4: 1.00, Low cohesion: 0

File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/NominalTypeDetector.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| DetectedNominal           | struct | 1     | 1       | 4          | high       |
| NominalTypeDetector       | class  | 2     | 6       | 2          | moderate   |
+---------------------------+--------+-------+---------+------------+------------+
Total: 2 classes, Average LCOM4: 1.50, Low cohesion: 0

File: /Users/fumiyatanaka/Work/OSS_cur/swift-complexity/Sources/SwiftComplexityCore/Analysis/SemanticLCOMCalculator.swift
Class Cohesion (LCOM4):
| Class/Struct              | Type   | LCOM4 | Methods | Properties | Cohesion   |
+---------------------------+--------+-------+---------+------------+------------+
| UnionFind                 | class  | 1     | 4       | 2          | high       |
| SemanticLCOMCalculator    | actor  | 1     | 11      | 2          | high       |
| PropertyAccessVisitor     | class  | 1     | 3       | 2          | high       |
| MethodCallVisitor         | class  | 1     | 2       | 2          | high       |

```

</p>
</details> 

### Code Quality

- [ ] Code follows the project's style guidelines
- [ ] Self-review of code completed
- [ ] Code is self-documenting with clear variable/function names
- [ ] Complex logic is commented where necessary

### Testing

- [ ] Tests added for new functionality
- [ ] Existing tests updated as needed
- [ ] All tests pass locally
- [ ] Edge cases considered and tested

### Documentation

- [ ] Documentation updated (if needed)
- [ ] README updated (if needed)
- [ ] CLAUDE.md updated (if architecture changes)
- [ ] API documentation updated (if applicable)

### Dependencies

- [ ] No new dependencies added
- [ ] New dependencies justified and documented
- [ ] Package.swift updated appropriately
- [ ] Cross-platform compatibility verified for new dependencies

### CI/CD

- [ ] CI pipeline passes (macOS & Linux)
- [ ] No new linting warnings introduced
- [ ] Code formatting validation passes (swift-format on macOS)
- [ ] Documentation linting passes (markdownlint)
- [ ] Self-complexity analysis in CI passes
- [ ] Release workflow compatibility maintained (if applicable)

## Additional Notes

(Any additional information that reviewers should know)

---

### Review Checklist

- [ ] **Complexity Analysis**: Results are reasonable and justified
- [ ] **Cross-Platform**: Changes work on both macOS and Linux
- [ ] **Test Coverage**: Adequate test coverage for new functionality
- [ ] **Documentation**: Updated appropriately for changes
- [ ] **CI/CD**: All automated checks pass
- [ ] **Architecture**: Changes align with project goals and patterns
- [ ] **Performance**: No unexpected performance regressions
- [ ] **Dependencies**: New dependencies are justified and secure

### Key Areas to Focus On

1. **Code Quality**: Maintainable, readable, and follows project conventions
2. **Platform Compatibility**: Proper handling of macOS/Linux differences
3. **Complexity Management**: New code doesn't unnecessarily increase complexity
4. **Test Quality**: Tests are comprehensive and meaningful
5. **Documentation**: Changes are properly documented for users and developers

### CODE REVIEW

File: Sources/SwiftComplexityCore/Analysis/ComplexityAnalyzer.swift [hunk 1]
```diff
@@ -9,16 +9,33 @@ public actor ComplexityAnalyzer: ComplexityAnalyzing {
...
     private let cognitiveCalculator: CognitiveComplexityCalculator
     private let functionDetector: FunctionDetector
+    private let nominalTypeDetector: NominalTypeDetector
+    private let lcomCalculator: SemanticLCOMCalculator?
+    private let enableLCOM4: Bool
 
-    public init() {
+    /// Initialize the analyzer
+    /// - Parameter projectRoot: Optional project root URL for LCOM4 analysis. If nil, LCOM4 will be disabled.
+    public init(projectRoot: URL? = nil) throws {
```
Comment (fummicc1) [inline Sources/SwiftComplexityCore/Analysis/ComplexityAnalyzer.swift]: [must]

because `projectRoot` cannot identify the exact location of `IndexStore`, ComplexityAnalyzer should request `indexStore` path as lcom4 enabled paramter.
